File: FrontEnd/src/modules/workout.ts
import { TimeHTMLAttributes } from 'vue';

export interface Workout {
  id: number;
  name: string;
  trainer: string;
  description: string;
  location: string;
  date: string;
  startTime: {hours: string; minutes:string;} | string;
  endTime: {hours: string; minutes:string;} | string;
  maxParticipants: number;
  nrOfParticipants: number;
}

export interface State {
  workouts: Workout[];
}

File: BackEnd/Model/Workout.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackEnd.Model
{
    [Table("workout")]
    public class Workout
    {
        [Column("id")]
        public int Id { get; init; }
        [Column("name")]
        public string? Name { get; init; }
        [Column("trainer")]
        public string? Trainer { get; init; }
        [Column("description")]
        public string? Description { get; init; }
        [Column("location")]
        public string? Location { get; init; }
        [Column("date")]
        public string? Date { get; init; }
        [Column("start_time")]
        public string? StartTime { get; init; }
        [Column("end_time")]
        public string? EndTime { get; init; }
        [Column("max_participants")]
        public int MaxParticipants { get; init; }
        [Column("nr_of_participants")]
        public int NrOfParticipants { get; init; }
        // public virtual ICollection<WorkoutUser> WorkoutUsers { get; set; } = new List<WorkoutUser>();
    }
}